pkg/apis/pipeline/v1beta1/param_types_test.go

```diff
@@ -171,6 +171,9 @@ func TestArrayOrString_UnmarshalJSON(t *testing.T) {
 		input  string
 		result v1beta1.ArrayOrString
 	}{
+		{"{\"val\": 123}", *v1beta1.NewArrayOrString("123")},
+		{"{\"val\": [1,2,3]}", *v1beta1.NewArrayOrString("[1,2,3]")},
+		{"{\"val\": [1,\"2\",3]}", *v1beta1.NewArrayOrString("[1,\"2\",3]")},
 		{"{\"val\": \"123\"}", *v1beta1.NewArrayOrString("123")},
 		{"{\"val\": \"\"}", *v1beta1.NewArrayOrString("")},
```

Comment:
> "{\"val\": }",

Yeah. Is it testable?

> 123 is a case starting with '1' not " or [

Ah I missed this test case. Nice!

Reply:
It will through error and it is expected. Since json won't unmarshal this right? I can add a test to assert error if that is necessary

Reply:
> It will through error and it is expected. Since json won't unmarshal this right? I can add a test to assert error if that is necessary

Yeah a test for an assert error would be good.

Reply:
I'm adding a test to assert error